### PR TITLE
[GSK-1069] Add text transform: remove accents

### PR DESCRIPTION
**Pull request by asselindebeauville**
## Description

I implemented `TextAccentRemovalTransformation` to remove accents with a perturbation `rate` that defaults to 1.0 (to remove all accents).

> [!NOTE]
> For the test, I used the texts and expected results from the issue (#1741). 
> Let me know if you want me to use the rate to ensure it is working appropriately.

## Related Issue

#1741 

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [x] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: tests/scan/test_text_transformations.py [hunk 1]
```diff
@@ -114,6 +114,31 @@ def test_punctuation_strip_transformation():
...
+
+    transformed = dataset.transform(t)
+    transformed_text = transformed.df.text.values
+
+    assert transformed_text[0] == "C'est l'ete"
+    assert transformed_text[1] == "ca et la"
+    assert transformed_text[2] == "Tieng Viet"
+    assert transformed_text[3] == "Etat"
+
+
```
Comment (mattbit) [inline tests/scan/test_text_transformations.py:141]: I would add an extra test string here to be sure we don't mess up with unicode

```suggestion
def test_accent_removal_transformation():
    dataset = _dataset_from_dict(
        {
            "text": [
                "C'est l'été",
                "çà et là",
                "Tiếng Việt",
                "État",
                "你好",
            ]
        }
    )

    from giskard.scanner.robustness.text_transformations import TextAccentRemovalTransformation

    t = TextAccentRemovalTransformation(column="text")

    transformed = dataset.transform(t)
    transformed_text = transformed.df.text.values

    assert transformed_text[0] == "C'est l'ete"
    assert transformed_text[1] == "ca et la"
    assert transformed_text[2] == "Tieng Viet"
    assert transformed_text[3] == "Etat"
    assert transformed_text[4] == "你好"


```
